### PR TITLE
[Rated disabilities] fix a11y issues with headers

### PR DESCRIPTION
**Pull request by jerekshoe**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#4784

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Desktop | before | after |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by samcoforma — APPROVED** [review]

File: src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx [hunk 1]
```diff
@@ -2,40 +2,26 @@ import React from 'react';
...
-    decisionText,
-    effectiveDate,
-    name,
-    ratingPercentage,
-  } = ratedDisability;
+  const { effectiveDate, name, ratingPercentage } = ratedDisability;
+
+  const ratingPercentageText =
+    typeof ratingPercentage === 'number' && ratingPercentage;
+  const headingText = [`${ratingPercentageText}%`, name].join(' ');
```
[the remark this answers is not in the document]
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx]: Should be fixed

File: src/applications/rated-disabilities/components/RatedDisabilityList.jsx [hunk 2]
```diff
@@ -112,22 +118,45 @@ const RatedDisabilityList = ({
...
   return (
     <div>
-      <h2 id="individual-ratings" className="vads-u-margin-y--1p5">
-        Your individual ratings
-      </h2>
+      <h3 className="vads-u-margin-y--2">Service-connected ratings</h3>
+      {sortToggle && (
+        <div id="ratings-sort-select-ab" className="vads-u-margin-bottom--2">
+          <SortSelect onSelect={setSortBy} sortBy={sortBy} />
+        </div>
```
Comment (Mottie) [inline src/applications/rated-disabilities/components/RatedDisabilityList.jsx:143]: I see this same code on lines 153-156; does that mean when you sort one, you sort both?

File: src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx [hunk 3]
```diff
@@ -1,41 +1,30 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+const getHeadingText = ratedDisability => {
+  const { name, ratingPercentage } = ratedDisability;
+
+  const headingParts = [name];
+  if (typeof ratingPercentage === 'number') {
+    headingParts.unshift(`${ratingPercentage}%`);
```
Comment (Mottie) [inline src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx:9]: I see `ratingPercentage` as a string in some mock data, so if it's a string, it won't get the percentage sign added?

**Review by Mottie — APPROVED** [review]

File: src/applications/rated-disabilities/components/RatedDisabilityList.jsx (hunk 2 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatedDisabilityList.jsx:143]: I may just need to remove the second one. This is behind a feature toggle that has never been turned on and I think that we probably just need to revisit the question of if there is any value in adding a sorting option here

File: src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx (hunk 3 above)
Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx:9]: I’ll go see what EVSS returns. It should always either be null, or a value (might be a string, not sure)

Comment (jerekshoe) [reply src/applications/rated-disabilities/components/RatedDisabilityListItem.jsx:9]: It should always be either null or a number, but I'm just going to check if it's not null to be safe. The mockdata should probably be updated

**Pull request merged**